core/lib/multivm/src/versions/vm_latest/utils/v26_upgrade.rs

```diff
@@ -0,0 +1,132 @@
+//! This module contains varios v26-upgrade-related functions to be used by the state keeper's
+//! unit tests.
+//! The correctness of the functionality present here is hard to enforce inside state keeper
+//! directly, so it is done inside unit tests of the multivm.
+
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+use ethabi::Token;
+use zksync_contracts::{l2_asset_router, l2_legacy_shared_bridge};
+use zksync_types::{utils::encode_ntv_asset_id, Address, StorageKey, H256, U256};
+use zksync_utils::env::Workspace;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct V26TestData {
+    pub l1_chain_id: U256,
+    pub l1_shared_bridge_address: Address,
+    pub l1_token_address: Address,
+    pub l2_token_address: Address,
+    pub l2_legacy_shared_bridge_address: Address,
+    pub l1_aliased_shared_bridge: Address,
+}
+
+pub fn get_test_data() -> V26TestData {
+    V26TestData {
+        l1_chain_id: 1.into(),
+        l1_shared_bridge_address: Address::from_str("abacabac00000000000000000000000000000001")
+            .unwrap(),
+        l1_token_address: Address::from_str("abacabac00000000000000000000000000000002").unwrap(),
+        l2_token_address: Address::from_str("de1b93723f716c741b771276465b7b58aa9c44a6").unwrap(),
+        l2_legacy_shared_bridge_address: Address::from_str(
+            "d316600b6009f94ab776e97f1fc985bc18b4e535",
+        )
+        .unwrap(),
+        l1_aliased_shared_bridge: Address::from_str("bcbdabac00000000000000000000000000001112")
+            .unwrap(),
+    }
+}
+
+fn get_test_v26_path(name: &str) -> PathBuf {
+    let core_path = Workspace::locate().core();
+    Path::new(&core_path).join(format!(
+        "lib/multivm/src/versions/testonly/v26_utils_outputs/{name}"
+    ))
+}
+
+pub fn trivial_test_storage_logs() -> HashMap<StorageKey, H256> {
+    let x: Vec<_> = serde_json::from_str(
+        &std::fs::read_to_string(get_test_v26_path("simple-test.json")).unwrap(),
+    )
+    .unwrap();
+    x.into_iter().collect()
+}
+
+pub fn post_bridging_test_storage_logs() -> HashMap<StorageKey, H256> {
+    let x: Vec<_> = serde_json::from_str(
+        &std::fs::read_to_string(get_test_v26_path("post-bridging.json")).unwrap(),
+    )
+    .unwrap();
+    x.into_iter().collect()
+}
+
+pub fn post_registration_test_storage_logs() -> HashMap<StorageKey, H256> {
+    let x: Vec<_> = serde_json::from_str(
+        &std::fs::read_to_string(get_test_v26_path("post-registration.json")).unwrap(),
+    )
+    .unwrap();
+    x.into_iter().collect()
+}
+
+fn empty_erc20_metadata() -> Vec<u8> {
+    ethabi::encode(&[
+        Token::Bytes(vec![]),
+        Token::Bytes(vec![]),
+        Token::Bytes(vec![]),
+    ])
+}
+
+pub fn encode_new_finalize_deposit(l1_chain_id: U256, l1_token_address: Address) -> Vec<u8> {
```

Comment:
Maybe a dumb suggestion regarding placing this test code (maybe, the entire module, or at least `encode_*deposit` methods). Maybe it makes sense to place it in `zksync_test_contracts` (e.g., as transactions produced by `TestAccount`) given that both `multivm` and `state_keeper` crates use test contracts, and the domain seems roughly right?

Reply:
I considered it, but these contracts rely on several imports from the `era-contracts` repo. If I place them fully into the `zksync_test_contracts`, it will require a giant amount of copy-paste from `era-contracts`

Reply:
FYI: I actually moved to this approach because of other reasons, but it appeared a lot easier than I thought initially should've done it from the start